packages/core/orchestration/src/transaction/transaction-orchestrator.ts

```diff
@@ -696,12 +700,28 @@ export class TransactionOrchestrator extends EventEmitter {
 
       if (!step.isCompensating()) {
         if (
-          step.definition.continueOnPermanentFailure &&
+          (step.definition.continueOnPermanentFailure ||
+            step.definition.skipOnPermanentFailure) &&
           !TransactionTimeoutError.isTransactionTimeoutError(timeoutError!)
         ) {
-          for (const childStep of step.next) {
-            const child = flow.steps[childStep]
-            child.changeState(TransactionStepState.SKIPPED_FAILURE)
+          if (step.definition.skipOnPermanentFailure) {
+            const until = isString(step.definition.skipOnPermanentFailure)
```

Comment:
q: should we validate the existence of this step id?

Reply:
Good push. Just added validation to it.